src/app/ConcreteAttributePath.h

```diff
@@ -37,7 +37,13 @@ struct ConcreteAttributePath
 
     bool operator==(const ConcreteAttributePath & other) const
     {
-        return mEndpointId == other.mEndpointId && mClusterId == other.mClusterId && mAttributeId == other.mAttributeId;
+        return (mEndpointId == other.mEndpointId) && (mClusterId == other.mClusterId) && (mAttributeId == other.mAttributeId);
+    }
+
+    bool operator<(const ConcreteAttributePath & path) const
+    {
+        return (mEndpointId < path.mEndpointId) || (!(mEndpointId < path.mEndpointId) && (mClusterId < path.mClusterId)) ||
+            (!(mClusterId < path.mClusterId) && (mAttributeId < path.mAttributeId));
```

Comment:
I would have expected:
```suggestion
        return (mEndpointId < path.mEndpointId) || ((mEndpointId == path.mEndpointId) && (mClusterId < path.mClusterId)) ||
            ((mClusterId == path.mClusterId) && (mAttributeId < path.mAttributeId));
```

because otherwise if `mEndpointId > path.mEndpointId` but `mClusterId < path.mClusterId` this will return true, which would be pretty surprising.  In particular, it would make this `operator<` non-transitive....